src/Controller/FormController.php

```diff
@@ -75,19 +75,158 @@ public function sendToSalesforce(Request $request)
+                    'formErrors' => $formErrors,
+                    'formData' => $formData,
+                ]);
+            } else {
+                // send to salesforce
+                $response = $this->client->request('POST', getenv('SALESFORCE_WEB_TO_CASE_URL'), [
+                    // these values are automatically encoded before including them in the URL
+                    'query' => $params->all(),
+                ]);
+        
+                if (!is_null($params->get('debug'))) {
+                    return new Response(
+                        $response->getContent()
+                    );
+                } elseif ($params->get('origin') == 'Website - Enquiry') {
+                    return $this->redirectToRoute('form_contact_thanks');
+                } else {
+                    return $this->redirectToRoute('form_contact_thanks_complaint');
+                }
+            }
+        }
+    }
+
+    public function validateForm(array $data)
+    {
+        $errorMessages = [
+            'nameErr' => [
+                'errors' => [],
+                'link' => '#name',
+            ],
+            'emailErr' => [
+                'errors' => [],
+                'link' => '#email',
+            ],
+            'phoneErr' => [
+                'errors' => [],
+                'link' => '#phone',
+            ],
+            'companyErr' => [
+                'errors' => [],
+                'link' => '#company',
+            ],
+            'jobTitleErr' => [
+                'errors' => [],
+                'link' => '#00Nb0000009IXEs',
+            ],
+            'postCodeErr' => [
+                'errors' => [],
+                'link' => '#post-code',
+            ],
+            'moreDetailErr' => [
+                'errors' => [],
+                'link' => '#more-detail',
+            ],
+        ];
+
+        // validation
+
+        // name
+        if (empty($data['name'])) {
+            $errorMessages['nameErr']['errors'][] = 'Enter your name';
+        }
+
+        if (strlen($data['name']) > 80) {
+            $errorMessages['nameErr']['errors'][] = 'Name must be 80 characters or fewer';
+        }
+
+        // email
+        if (empty($data['email']) || !filter_var($data['email'], FILTER_VALIDATE_EMAIL)) {
+            $errorMessages['emailErr']['errors'][] = 'Enter an email address in the correct format, like name@example.com';
+        }
+
+        if (strlen($data['email']) > 80) {
+            $errorMessages['emailErr']['errors'][] = 'Email address must be 80 characters or fewer';
+        }
+        
+        // phone
+        if (empty($data['phone'])) {
+            $errorMessages['phoneErr']['errors'][] = 'Enter a telephone number, like 01632 960 001, 07700 900 982 or +44 0808 157 0192';
+        }
+
+        if (strlen($data['phone']) > 20) {
+            $errorMessages['phoneErr']['errors'][] = 'Telephone number must be 20 characters or fewer';
+        }
```

Comment:
Has someone actually told us that both email and telephone number are required? 
Should we have either/or?

Reply:
This validation was again based on the HTML5 validation. Email will certainly be required for Pardot but telephone we can leave as optional. Although, there is an checkbox for a callback which would make the telephone a requirement I believe.